tests/test_simulation.py

```diff
@@ -416,11 +418,452 @@ def test_element_nodal_forces_elemental(self, allkindofcomplexity):
+
+    def test_structural_temperature(self, transient_simulation):
+        result = transient_simulation.structural_temperature(time_step_ids=[2])
+        assert len(result._fc) == 1
+        assert result._fc.get_time_scoping().ids == [2]
+        field = result._fc[0]
+        op = transient_simulation._model.operator("BFE")
+        field_ref = op.eval()[0]
+        assert field.component_count == 1
+        assert np.allclose(field.data, field_ref.data)
+
+    def test_structural_temperature_nodal(self, transient_simulation):
+        result = transient_simulation.structural_temperature_nodal(time_step_ids=[2])
+        assert len(result._fc) == 1
+        assert result._fc.get_time_scoping().ids == [2]
+        field = result._fc[0]
+        op = transient_simulation._model.operator("BFE")
+        op.connect(9, core.locations.nodal)
+        field_ref = op.eval()[0]
+        assert field.component_count == 1
+        assert np.allclose(field.data, field_ref.data)
+
+    def test_structural_temperature_elemental(self, transient_simulation):
+        result = transient_simulation.structural_temperature_elemental(
+            time_step_ids=[2]
+        )
+        assert len(result._fc) == 1
+        assert result._fc.get_time_scoping().ids == [2]
+        field = result._fc[0]
+        op = transient_simulation._model.operator("BFE")
+        op.connect(9, core.locations.elemental)
+        field_ref = op.eval()[0]
+        assert field.component_count == 1
+        assert np.allclose(field.data, field_ref.data)
+
+    def test_element_nodal_forces(self, allkindofcomplexity):
+        transient_simulation = dpf.load_simulation(
+            data_sources=allkindofcomplexity,
+            simulation_type=AvailableSimulationTypes.transient_mechanical,
+        )
+        result = transient_simulation.element_nodal_forces(time_step_ids=[1])
+        assert len(result._fc) == 1
+        assert result._fc.get_time_scoping().ids == [1]
+        field = result._fc[0]
+        op = transient_simulation._model.operator("ENF")
+        field_ref = op.eval()[0]
+        assert field.component_count == 3
+        assert np.allclose(field.data, field_ref.data)
+
+    def test_element_nodal_forces_nodal(self, allkindofcomplexity):
+        transient_simulation = dpf.load_simulation(
+            data_sources=allkindofcomplexity,
+            simulation_type=AvailableSimulationTypes.transient_mechanical,
+        )
+        result = transient_simulation.element_nodal_forces_nodal(time_step_ids=[1])
+        assert len(result._fc) == 3
+        assert result._fc.get_time_scoping().ids == [1]
+        field = result._fc[0]
+        op = transient_simulation._model.operator("ENF")
+        op.connect(9, core.locations.nodal)
+        field_ref = op.eval()[0]
+        assert field.component_count == 3
+        assert np.allclose(field.data, field_ref.data)
+
+    def test_element_nodal_forces_elemental(self, allkindofcomplexity):
```

Comment:
I see that "named_selection" and "selection" keywords are not tested yet. is it because it's a work in progress?
Should we maybe add a code coverage bot to help?

Reply:
Just added a test for named_selection.
As for the Selection, the Selection object is not ready yet so no test for it. It is indeed a work in progress.